app/src/main/java/ch/epfl/sweng/rps/models/User.kt

```diff
@@ -4,33 +4,37 @@ package ch.epfl.sweng.rps.models
  *
  */
 data class User(
-    val username: String?,
+    val username: String? = "",
 
-    val uid: String,
+    val uid: String = "",
 
-    val gamesHistoryPrivacy: Privacy,
+    val gamesHistoryPrivacy: String = "",
 
-    val hasProfilePhoto: Boolean,
+    val hasProfilePhoto: Boolean = false,
 
-    val email: String?
+    val email: String? = "",
+
+    val matchesList: List<String>? = null
 ) {
 
     fun deepCopy(
         username: String? = this.username,
         uid: String = this.uid,
-        gamesHistoryPrivacy: Privacy = this.gamesHistoryPrivacy,
+        gamesHistoryPrivacy: String = this.gamesHistoryPrivacy,
         hasProfilePhoto: Boolean = this.hasProfilePhoto,
-        email: String? = this.email
+        email: String? = this.email,
+        matchesList: List<String>? = this.matchesList
     ): User {
-        return User(username, uid, gamesHistoryPrivacy, hasProfilePhoto, email)
+        return User(username, uid, gamesHistoryPrivacy, hasProfilePhoto, email, matchesList)
     }
 
     enum class Field(val value: String) {
         USERNAME("username"),
         GAMES_HISTORY_PRIVACY("games_history_privacy"),
         HAS_PROFILE_PHOTO("has_profile_photo"),
         EMAIL("email"),
-        // UID("uid")
+        UID("uid"),
+        MATCHESLIST("matchesList")
```

Comment:
What is the use case for Field? The name is not descriptive.

Reply:
UID is the unique id that firebase gives to each authenticated user, it is here for query optimizations inside firebase.